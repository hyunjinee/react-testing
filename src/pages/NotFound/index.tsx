import styled from 'styled-components';

const NotFound: React.FC = () => {
  return <Container>page not found</Container>;
};

const Container = styled.div`
  font-size: 20px;
`;

export default NotFound;
