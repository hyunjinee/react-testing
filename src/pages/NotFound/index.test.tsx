import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import NotFound from '.';

describe('<NotFound/>', () => {
  it('render component correctly', () => {
    const { container } = render(<NotFound />);

    const message = screen.getByText('page not found');
    expect(message).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
