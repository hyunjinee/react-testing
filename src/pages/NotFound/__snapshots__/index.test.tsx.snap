// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/> render component correctly 1`] = `
.c0 {
  font-size: 20px;
}

<div>
  <div
    class="c0"
  >
    page not found
  </div>
</div>
`;
